File: web/src/types/vehicle.ts
/* ========================================
   车辆模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** 车辆详情（对齐 Vehicle schema） */
export interface Vehicle {
  id: number;
  plateNo: string;
  defaultDriverUserId?: number | null;
  defaultDriverOther?: string | null;
  defaultDriverName?: string | null;
  defaultEscortUserId?: number | null;
  defaultEscortOther?: string | null;
  defaultEscortName?: string | null;
  remark?: string | null;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** 车辆列表项（对齐 VehicleSummary schema） */
export interface VehicleSummary {
  id: number;
  plateNo: string;
  defaultDriverName?: string | null;
  defaultEscortName?: string | null;
  status: 0 | 1;
  createdAt: string;
}

/** 车辆列表分页响应 */
export interface VehicleListResponse {
  items: VehicleSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建车辆请求 */
export interface CreateVehicleRequest {
  plateNo: string;
  defaultDriverUserId?: number | null;
  defaultDriverOther?: string | null;
  defaultEscortUserId?: number | null;
  defaultEscortOther?: string | null;
  remark?: string;
  status?: 0 | 1;
}

/** 更新车辆请求（常态班组整体覆盖，可清空以切换 用户 ↔ OTHER） */
export interface UpdateVehicleRequest {
  plateNo?: string;
  defaultDriverUserId?: number | null;
  defaultDriverOther?: string | null;
  defaultEscortUserId?: number | null;
  defaultEscortOther?: string | null;
  remark?: string;
}

/** 车辆查询参数 */
export interface VehicleQueryParams {
  keyword?: string;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}

File: api/src/main/java/com/aistore/module/vehicle/entity/Vehicle.java
package com.aistore.module.vehicle.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.FieldStrategy;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 车辆实体（表 vehicle）。
 * 常态化司机/跟车员为“软引用 sys_user 或 OTHER 文本替补”二选一；
 * 这四个字段标注 updateStrategy=IGNORED，使更新时可写入 null（支持 用户 ↔ OTHER 的切换/清空）。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("vehicle")
public class Vehicle {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private String plateNo;

    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private Long defaultDriverUserId;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private String defaultDriverOther;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private Long defaultEscortUserId;
    @TableField(updateStrategy = FieldStrategy.IGNORED)
    private String defaultEscortOther;

    private String remark;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}
